pypesto/objective/priors.py

```diff
@@ -0,0 +1,264 @@
+    """
+
+    def __init__(self,
+                 prior_list: list):
+
+        self.prior_list = prior_list
+
+        super().__init__(fun=self.density_for_full_parameter_vector,
+                         grad=self.gradient_for_full_parameter_vector,
+                         hess=self.hessian_for_full_parameter_vector,
+                         hessp=self.hessian_vp_for_full_parameter_vector)
+
+    def __deepcopy__(self, memodict={}):
+        other = ParameterPriors(deepcopy(self.prior_list))
+        return other
+
+    def density_for_full_parameter_vector(self, x):
+
+        density_val = 0
+        for prior in self.prior_list:
+            density_val -= prior['density_fun'](x[prior['index']])
+
+        return density_val
+
+    def gradient_for_full_parameter_vector(self, x):
+
+        grad = np.zeros_like(x)
+
+        for prior in self.prior_list:
+            grad[prior['index']] -= prior['density_dx'](x[prior['index']])
+
+        return grad
+
+    def hessian_for_full_parameter_vector(self, x):
+
+        hessian = np.zeros((len(x), len(x)))
+
+        for prior in self.prior_list:
+            hessian[prior['index'], prior['index']] -= \
+                prior['density_ddx'](x[prior['index']])
+
+        return hessian
+
+    def hessian_vp_for_full_parameter_vector(self, x, p):
+
+        h_dot_p = np.zeros_like(p)
+
+        for prior in self.prior_list:
+            h_dot_p[prior['index']] -= \
+                prior['density_ddx'](x[prior['index']]) * p[prior['index']]
+
+        return h_dot_p
+
+
+def get_parameter_prior_dict(index: int,
+                             prior_type: str,
+                             prior_parameters: list,
+                             parameter_scale: str = 'lin'):
+
+    """
+    Returns the prior dict used to define priors for some default priors.
+
+    index:
+        index of the parameter in x_full
+
+    prior_type: str
+        Prior is defined in LINEAR parameter space! prior_type can from
+        {uniform, normal, laplace, logUniform, logNormal, logLaplace}
+
+    prior_parameters:
+        Parameters of the priors. Parameters are defined in linear scale.
+
+    parameter_scale:
+        scale, in which parameter is defined (since a parameter can be
+        log-transformed, while the prior is always defined in the linear space)
+    """
+
+    log_f, d_log_f_dx, dd_log_f_ddx = \
+        _prior_densities(prior_type, prior_parameters)
+
+    if parameter_scale == 'lin':
+
+        return {'index': index,
+                'density_fun': log_f,
+                'density_dx': d_log_f_dx,
+                'density_ddx': dd_log_f_ddx}
+
+    elif parameter_scale == 'log':
+
+        def log_f_log(x_log):
+            """log-prior for log-parameters"""
+            return log_f(math.exp(x_log))
+
+        def d_log_f_log(x_log):
+            """derivative of log-prior w.r.t. log-parameters"""
+            return d_log_f_dx(math.exp(x_log)) * math.exp(x_log)
```

Comment:
the math functions only work on scalars, to my knowledge, thus one should rather use e.g. `np.exp` here. this would also directly allow to use the code for vectorization, i.e. when multiple parameters share a prior.

```python
>>> import math
>>> import numpy as np
>>> a = np.array([0.1, 0.2, 0.3])
>>> math.exp(a)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
TypeError: only size-1 arrays can be converted to Python scalars
>>> np.exp(a)
array([1.10517092, 1.22140276, 1.34985881])
```